runtime/near-vm-logic/src/logic.rs

```diff
@@ -948,6 +948,146 @@ impl<'a> VMLogic<'a> {
     }
 
+    /// Hashes the given value using RIPEMD-160 and returns it into `register_id`.
+    ///
+    /// # Errors
+    ///
+    /// If `value_len + value_ptr` points outside the memory or the registers use more memory than
+    /// the limit with `MemoryAccessViolation`.
+    ///
+    /// # Cost
+    ///
+    /// `base + write_register_base + write_register_byte * num_bytes + ripemd160_base + ripemd160_byte * num_bytes`
+    #[cfg(feature = "protocol_feature_evm")]
+    pub fn ripemd160(&mut self, value_len: u64, value_ptr: u64, register_id: u64) -> Result<()> {
+        self.gas_counter.pay_base(ripemd160_base)?;
+        let value = self.get_vec_from_memory_or_register(value_ptr, value_len)?;
+        self.gas_counter.pay_per_byte(ripemd160_byte, value.len() as u64)?;
+
+        use ripemd160::Digest;
+
+        let value_hash = ripemd160::Ripemd160::digest(&value);
+        self.internal_write_register(register_id, value_hash.as_slice().to_vec())
+    }
+
+    /// Hashes the given value using BLAKE2b and returns it into `register_id`.
+    ///
+    /// # Errors
+    ///
+    /// If `value_len + value_ptr` points outside the memory or the registers use more memory than
+    /// the limit with `MemoryAccessViolation`.
+    ///
+    /// # Cost
+    ///
+    /// `base + write_register_base + write_register_byte * num_bytes + blake2b_base + blake2b_byte * num_bytes`
+    #[cfg(feature = "protocol_feature_evm")]
+    pub fn blake2b(&mut self, value_len: u64, value_ptr: u64, register_id: u64) -> Result<()> {
+        use blake2::{Blake2b, Digest};
+
+        self.gas_counter.pay_base(blake2b_base)?;
+        let value = self.get_vec_from_memory_or_register(value_ptr, value_len)?;
+        self.gas_counter.pay_per_byte(blake2b_byte, value.len() as u64)?;
+
+        let mut hasher = Blake2b::new();
+        hasher.update(&value);
+        let value_hash = hasher.finalize();
+        self.internal_write_register(register_id, value_hash.as_slice().to_vec())
+    }
+
+    /// The compression function of the blake2 algorithm.
+    ///
+    /// Takes as an argument the state vector "h", message block vector "m" (the
+    /// last block is padded with zeros to full block size, if required), 2w-bit
+    /// offset counter "t", and final block indicator flag "f". Local vector
+    /// v[0..15] is used in processing. F returns a new state vector. The number
+    /// of rounds, "r", is 12 for BLAKE2b and 10 for BLAKE2s. Rounds are
+    /// numbered from 0 to r - 1.
+    ///
+    /// # Cost
+    /// base + write_register_base + write_register_byte * 64 + blake2b_f_base
+    #[cfg(feature = "protocol_feature_evm")]
+    pub fn blake2b_f(
+        &mut self,
+        rounds_ptr: u64,
+        h_ptr: u64,
+        m_ptr: u64,
+        t_ptr: u64,
+        f_ptr: u64,
+        register_id: u64,
+    ) -> Result<()> {
+        use std::convert::TryFrom;
+
+        let rounds = self.memory_get_u32(rounds_ptr)?;
```

Comment:
Is there a reason we pass `rounds` and `f` by pointer here, while we pass `v`  by value in `ecrecover`. My understanding of this specific part of code is very vague, but, intuitively, if we can pass by value, we should?

Reply:
`v` is just single byte (`u8`).

Reply:
`f` is also a single byte (`bool`), and `rounds` is a single `u32`. I still don't see a reason to not just pass them directly, instead of passing pointers to them (which doesn't mean that there isn't such a reason).

Reply:
Sure, makes sense to me. @joshuajbouw?